### DISCUSSION TITLE
Add metadata from other keys to MetaTensor.meta

### DISCUSSION THREAD

**Discussion by tom-osika** [Project-MONAI/MONAI#6350] [Q&A]
I'm using a dataloader with the `LoadImaged` transform (with `image_only=True`) to read in `MetaTensor` objects. The data dictionary yielded by the data loader looks like the following, for example:

```
{'image': tensor([[[ 5,  2,  9],
          [61, 39,  4],
          [56, 32, 13]]]),
 'other_meta': {'meta1': 'val1', 'meta2': 'val2'}}
```
Where the `tensor` is a `MetaTensor` that contains other metadata such as the affine, etc. Is there any way to combine the `other_meta` dictionary with the metadata in `MetaTensor.meta` using transforms? There are no overlapping keys between the two dictionaries.

Thanks!

**Comment by KumoLiu** [comment]
Hi @tao558, you could try to use `ToMetaTensord`.
https://github.com/Project-MONAI/MONAI/blob/1a55ba5423d04d2ef7ac19356ccabc4c7906f577/monai/transforms/meta_utility/dictionary.py#L89-L94

Hope it can help you, thanks!

**Reply by tom-osika** [reply]
Hi @KumoLiu, thank you for your response.

I tried using `ToMetaTensord` initially. If I set `image_only=False` when loading the images, I'll get something like the following each iteration before the `ToMetaTensord`:

```
{'image': tensor([[[ 5,  2,  9],
          [61, 39,  4],
          [56, 32, 13]]]),
 'image_meta_dict': {'affine': torch.Tensor(...), ...} # from reading the image with LoadImaged
 'other_meta': {'meta1': 'val1', 'meta2': 'val2'}} # My metadata
```

With `ToMetaTensord`, I can put the `image_meta_dict` into the `MetaTensor`, but not both meta dictionaries. Am I missing something?

Thanks again.

**Reply by KumoLiu** [reply]
Hi @tao558, you could combine these two meta info like this:
```
def combine_meta(data):
    data["image_meta_dict"].update(data["other_meta"])
    return data

out = Lambda(func=lambda x: combine_meta(x))(data)
```

Hope it can help you, thanks!
